modules/RTC/TraceablePeerConnection.js

```diff
@@ -2571,6 +2573,27 @@ TraceablePeerConnection.prototype.setSenderVideoConstraints = function(frameHeig
     if (!localVideoTrack || localVideoTrack.isMuted()) {
         return Promise.resolve();
     }
+
+    const prevPromise = this._lastVideoSenderUpdatePromise;
+    const nextPromise = Promise.allSettled([ prevPromise ])
+        .then(() => this.updateVideoSenderParameters(frameHeight, localVideoTrack));
+
+    this._lastVideoSenderUpdatePromise = nextPromise;
+
+    return nextPromise;
+};
+
+/**
+ * Updating video sender parameters as wrapped-up promise to have ability
+ * to chain promises sequentially later and avoid problem with transaction id
+ * which is reset at the end of video sender setParameters method and can affect
+ * next on-fly update
+ *
+ * @param {number} frameHeight - The max frame height to be imposed on the outgoing video stream.
+ * @param {JitsiLocalTrack} - The local track for which the sender constraints have to be applied.
+ * @returns {Promise} promise that will be resolved when the operation is successful and rejected otherwise.
+ */
+TraceablePeerConnection.prototype.updateVideoSenderParameters = function(frameHeight, localVideoTrack) {
```

Comment:
I'd suspect this should be private.

Reply:
added underscore prefix here for making it private.